pymc3/tests/test_data_container.py

```diff
@@ -40,19 +40,30 @@ def test_sample(self):
 
             trace = pm.sample(1000, init=None, tune=1000, chains=1)
             pp_trace0 = pm.sample_posterior_predictive(trace, 1000)
+            pp_trace01 = pm.fast_sample_posterior_predictive(trace, 1000)
 
             x_shared.set_value(x_pred)
             pp_trace1 = pm.sample_posterior_predictive(trace, samples=1000)
+            pp_trace11 = pm.fast_sample_posterior_predictive(trace, samples=1000)
             prior_trace1 = pm.sample_prior_predictive(1000)
 
         assert prior_trace0['b'].shape == (1000,)
         assert prior_trace0['obs'].shape == (1000, 100)
+        assert prior_trace1['obs'].shape == (1000, 200)
```

Comment:
Why does this have a different shape than the one above?

Reply:
This happens because the shape of the predictor variable (`x_shared`) has changed.  It was originally a vector of size 100 (line 36), but on line 45, we update it to be of size 200 (see line 33 for the new value).  This changes the size of the prior predictive trace.

This was always true, I just added a check to verify it.

Reply:
It has a different shape because of the change to the predictor variable.  When we reset the `Data` item, we change the shape of the predictor, and so we change the shape of the observed variable.

This is not a change in the function of the code -- I just tested it slightly more thoroughly to verify the prior predictive sampling was working properly (since posterior predictive sampling did not, and does not -- when you change the shape of the predictors, in general, you must get a new model and may need to edit the posterior trace.).